docs/docs/adding-search-with-algolia.md

```diff
@@ -2,7 +2,604 @@
+              </Results>
+            </Index>
+          ))}
+          <By>
+            Powered by{' '}
+            <a href="https://www.algolia.com">
+              <Algolia size="1em" /> Algolia
+            </a>
+          </By>
+        </HitsWrapper>
+      </InstantSearch>
+    )
+  }
+}
+```
+
+There's quite a lot happening in this file so let's break it down piece by piece.
+
+```js
+import {
+  InstantSearch,
+  Index,
+  Hits,
+  connectStateResults,
+} from 'react-instantsearch-dom'
+```
+
+`InstantSearch` imported from [`react-instantsearch-dom`](https://community.algolia.com/react-instantsearch) is Algolia's main off-the-shelf React component that allows your whole search experience to connect to their service. As the name suggests, `Index` allows you to tap into an individual index and `Hits` provides you with the data returned for a user's search input. Finally [`connectStateResults`](https://community.algolia.com/react-instantsearch/connectors/connectStateResults.html) provides high-level stats about the current search state such as the query, the number of results and how long it took to fetch them.
+
+Next we import the styled components that make up the UI.
+
+```js
+import { Algolia } from 'styled-icons/fa-brands/Algolia'
+import { Root, SearchBox, HitsWrapper, By } from './styles'
+```
+
+Notice that we're importing the Algolia logo from `styled-icons`. That's because if you're using Algolia's generous free tier, they ask you to acknowledge them with a little `Powered by Algolia` link below the search results. `react-instantsearch-dom` provides a [`PoweredBy` component](https://community.algolia.com/react-instantsearch/widgets/PoweredBy.html) for that but you can of course build and style your own (as done here).
+
+We'll get back to those styled components once we're done with `index.js`. For now, let's move on.
+
+```js
+import Input from './Input'
+```
+
+The Input component is where the user enters the search string. It is quite short since the grunt work is done by one of Algolia's components called [`connectSearchBox`](https://community.algolia.com/react-instantsearch/connectors/connectSearchBox.html):
+
+```jsx
+// src/components/Search/Input.js
+import React from 'react'
+import { connectSearchBox } from 'react-instantsearch-dom'
+
+import { Loupe, Form, Input } from './styles'
+
+export default connectSearchBox(({ refine, ...rest }) => (
+  <Form>
+    <Input
+      type="text"
+      placeholder="Search"
+      aria-label="Search"
+      onChange={e => refine(e.target.value)}
+      {...rest}
+    />
+    <Loupe />
+  </Form>
+))
+```
+
+Let's again worry about the styled components `Loupe`, `Form`, `Input` later.
+
+The next line imports hit components for every type of result we want to display to the user. The hit component determines how attributes of matching results such as author and title in the case of a blog post are displayed to the user.
```

Comment:
The term "hit" was new to me when I first used Algolia. Maybe it's worth an explicit definition that a "hit" is what Algolia refers to what I would call a "search result"?